### PR TITLE
TimestampMetadataValue

### PR DESCRIPTION
**Pull request by sryza**
## Summary & Motivation

Introduces a `TimestampMetadataValue` type, that wraps a Unix timestamp. The advantage of using this metadata type instead of a float is that the frontend can display it as a timezone instead of as a bug number.  Does not contain a timezone, so the frontend can display it in a timezone that's consistent with the rest of the timezones on the frontend.

Issue: https://github.com/dagster-io/dagster/issues/13344

### CODE REVIEW

**Comment by hellendag** [comment]
The TypeScript errors [here](https://buildkite.com/dagster/dagster-dagster/builds/77980#018e1c44-d4bd-437a-b66e-0a0ee78f305b/211-429) are legit, you'll need to update the relevant switch statements to include the new type.

**Review by hellendag — CHANGES_REQUESTED** [review]
Just requesting the changes to fix the TS errors. Otherwise seems fine to me. Let me know if I can be of help getting the UI in place.

File: python_modules/dagster/dagster/_core/definitions/metadata/__init__.py [hunk 1]
```diff
@@ -406,6 +408,30 @@ def emit_metadata(context, df):
...
+    @public
+    @staticmethod
+    def timestamp(value: Union["float", datetime]) -> "TimestampMetadataValue":
+        """Static constructor for a metadata value wrapping a UNIX timestamp as a
+        :py:class:`TimestampMetadataValue`. Can be used as the value type for the `metadata`
+        parameter for supported events.
+
+        Args:
+            value (Union[float, datetime]): The unix timestamp value for a metadata entry. If a
+                datetime is provided, the timestamp will be extracted. datetimes without timestamps
```
[the remark this answers is not in the document]
Comment (sryza) [reply python_modules/dagster/dagster/_core/definitions/metadata/__init__.py:420]: thanks for catching

**Comment by sryza** [comment]
>  Let me know if I can be of help getting the UI in place.

Can I take you up on this? Made a change that I thought would fix it, but still seeing an error. And not sure about how to get them to show up as formatted datetimes.

**Review by hellendag — APPROVED** [review]

**Pull request merged**